GameServerLib/GameObjects/Spells/BuffManager.cs

```diff
@@ -0,0 +1,410 @@
+﻿using GameServerCore;
+using GameServerCore.Domain;
+using GameServerCore.Domain.GameObjects;
+using GameServerCore.Enums;
+using LeagueSandbox.GameServer;
+using LeagueSandbox.GameServer.GameObjects.Spells;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using PacketDefinitions420.PacketDefinitions.S2C;
+using Priority_Queue;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace GameServerLib.GameObjects.Spells
+{
+    class BuffManager : IBuffManager
+    {
+        //private Dictionary<string, IBuff> _buffs;
+        private SimplePriorityQueue<IBuff, float> _buffQueue;
+        private IBuff[] _slots;
+        private Game _game;
```

Comment:
If you can find a way to avoid this `Game` class get copied everywhere, please do it. I have a suggestion to just copy `PacketNotifier` this time.